### PR TITLE
Fix several E2E tests errors following refactor merge

### PR DESCRIPTION
**Pull request by vboulineau**
### What does this PR do?

Requires https://github.com/DataDog/test-infra-definitions/pull/555
Gitlab Runner image id will be changed when merged.

Fix missing changes for export in `containers` test.
Fix bad copy/paste in Dogstatsd install script test

### Motivation

Fix failing E2E

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by KevinFairise2 — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]

File: test/new-e2e/pkg/e2e/suite.go [hunk 1]
```diff
@@ -283,7 +283,7 @@ func (bs *BaseSuite[Env]) reconcileEnv(targetProvisioners ProvisionerMap) {
 		}
 
 		if err != nil {
-			panic(fmt.Errorf("unable to provision stack: %s, provisioner %s, err: %v", bs.params.stackName, id, err))
+			panic(fmt.Errorf("your stack '%s' provisioning failed, check logs above. Provisioner was %s, failed with err: %v", bs.params.stackName, id, err))
```
Comment (pducolin) [inline test/new-e2e/pkg/e2e/suite.go:286]: **❓ question**
If we panic here, doesn't this prevent cleaning up a failing deployment ?

File: test/new-e2e/tests/containers/ecs_test.go [hunk 2]
```diff
@@ -51,11 +51,16 @@ func (suite *ecsSuite) SetupSuite() {
...
 	}
 
-	_, stackOutput, err := infra.GetStackManager().GetStack(ctx, "ecs-cluster", stackConfig, ecs.Run, false)
+	_, stackOutput, err := infra.GetStackManager().GetStackNoDeleteOnFailure(ctx, "ecs-cluster", stackConfig, ecs.Run, false, nil)
 	suite.Require().NoError(err)
 
-	fakeintakeHost := stackOutput.Outputs["fakeintake-host"].Value.(string)
-	suite.Fakeintake = fakeintake.NewClient(fmt.Sprintf("http://%s", fakeintakeHost))
+	fakeintake := &components.FakeIntake{}
+	fiSerialized, err := json.Marshal(stackOutput.Outputs["dd-Fakeintake-aws-ecs"].Value)
```
Comment (pducolin) [inline test/new-e2e/tests/containers/ecs_test.go:58]: **❓ question**
Is capital `F` on purpose ?

**Comment by KevinFairise2** [comment]
I think you need to update the `TEST_INFRA_DEFINITIONS_BUILDIMAGE` in [.gitlab-ci.yml](https://github.com/DataDog/datadog-agent/blob/main/.gitlab-ci.yml#L182)

File: test/new-e2e/pkg/e2e/suite.go (hunk 1 above)
Comment (pducolin) [reply test/new-e2e/pkg/e2e/suite.go:286]: What if you return the error and let the caller decide how to handle it ?

Comment (vboulineau) [reply test/new-e2e/pkg/e2e/suite.go]: Yep that's what I pushed, but clean implementation requires larger changes, so we're back at provisioning in `BeforeTest`.

File: test/new-e2e/tests/containers/ecs_test.go (hunk 2 above)
Comment (vboulineau) [reply test/new-e2e/tests/containers/ecs_test.go:58]: Yes, the types are exported "as-is" from `test-infra` (so with capitals if any)

**Review by pducolin — APPROVED** [review]

**Pull request merged**